src/i_oplmusic.c

```diff
@@ -1413,7 +1415,19 @@ static void TrackTimerCallback(void *arg)
 
         if (running_tracks <= 0 && song_looping)
         {
-            OPL_SetCallback(5000, RestartSong, NULL);
+            if (song_looping > 2)
+            {
+                --song_looping;
+            }
+
+            if (song_looping == 2)
+            {
+                OPL_SetCallback(3000000, FinishSong, NULL);
```

Comment:
And why this?

Reply:
I put a delay of 3 seconds to pad the song with a bit of silence. Some music players cut off the very end of songs. The `FinishSong` callback is needed so the loop actually ends. I'm not sure why, but without explicitly telling the OPL driver to stop the song, `I_MusicIsPlaying` always returns true.